packages/clay-date-picker/src/DayNumber.tsx

```diff
@@ -41,11 +51,28 @@ const ClayDatePickerDayNumber: React.FunctionComponent<IProps> = ({
 				'yyyy MM dd'
 			)}
 			className={classNames}
-			disabled={day.outside}
-			onClick={() => onClick(day.date)}
+			disabled={outside}
+			onClick={() => onClick(date)}
+			onKeyDown={(event) => {
+				// When tabbing and selecting a DayNumber using
+				// SPACE key the active state it's not being removed.
+				// See https://github.com/liferay/clay/issues/3374 for more details.
+				if (event.keyCode === KEY_CODE_SPACE) {
+					event.preventDefault();
+				}
+
+				onKeyDown(event);
+			}}
+			onKeyUp={(event) => {
+				if (event.keyCode === KEY_CODE_SPACE) {
+					onClick(date);
+				}
+
+				onKeyUp(event);
+			}}
```

Comment:
One question: this just to retrieve the `space` event that we gave `preventDefault` before?

Reply:
Yes, when keyDown, We're preventing and when SPACE key is released, onClick cb will be called.